### PR TITLE
Cpu refactoring

### PR DESCRIPTION
**Pull request by ozgunozerk**

### CODE REVIEW

**Review by nazar-pc — CHANGES_REQUESTED** [review]
Please move all CPU-related code into `cpu` folder, not just two Rust files.

**Comment by ozgunozerk** [comment]
> Please move all CPU-related code into `cpu` folder, not just two Rust files.

Yes, we did that with the cuda-ptx branch. I reflected the same changes to this branch now, as it should have been :)

File: implementation.md [hunk 1]
```diff
@@ -52,10 +52,10 @@ implementation.
 ADX code path is selected at run time. For testing purposes assembly
 support can be suppressed with `--features no-asm` at cargo command
 line. Otherwise `build.rs` adds `assembly.S` or
-`win64/mod256-189-x86_64.asm` to build sequence depending on whether or
+`cpu.win64/mod256-189-x86_64.asm` to build sequence depending on whether or
 not one uses a non-Microsoft or Microsoft compiler. In former case the
-`assembly.S` makes further choice among `coff` (used by MinGW), `elf` (used
-by Linux, *BSD, Solaris) or `mach-o` (used by MacOS) executable formats.
+`assembly.S` makes further choice among `cpu.coff` (used by MinGW), `cpu.elf` (used
+by Linux, *BSD, Solaris) or `cpu.mach-o` (used by MacOS) executable formats.
```
Comment (nazar-pc) [inline implementation.md:58]: Please fix these

**Review by nazar-pc — COMMENTED** [review]
Looks good otherwise

**Review by nazar-pc — APPROVED** [review]

**Pull request merged**